File: selfstudy/golang/books_sum.go
/*
task 251:
Books - Sum
You need to calculate the total sum of book page counts
*/

package main

import (
	"fmt"
	"reflect"
)

func TotalBooks(pages []int) int {
	sum := 0
	for _, v := range pages {
		sum += v
	}
	return sum
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, TotalBooks([]int{10, 20, 30})},
		{[]int{50, 60, 70}, TotalBooks([]int{50, 60, 70})},
		{[]int{1, 2, 3}, TotalBooks([]int{1, 2, 3})},
		{[]int{100, 200}, TotalBooks([]int{100, 200})},
		{[]int{}, TotalBooks([]int{})},
	}
	
	for idx, t := range tests {
		res := TotalBooks(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done

File: selfstudy/js/books_sum.js
/*
task 51:
Books - Sum
You need to calculate the total sum of book page counts
*/

function total_books(pages) {
  return pages.reduce((a, b) => a + b, 0);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: total_books([10, 20, 30]) },
  { input: [50, 60, 70], expected: total_books([50, 60, 70]) },
  { input: [1, 2, 3], expected: total_books([1, 2, 3]) },
  { input: [100, 200], expected: total_books([100, 200]) },
  { input: [], expected: total_books([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('total_books', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(total_books(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, total_books(t.input) === t.expected || JSON.stringify(total_books(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// done
